tests/test-ansible-functional.sh

```diff
@@ -154,7 +154,9 @@ function enable_maas_api {
   #                  different virtualenv which are done in three tasks. These three tasks are being
   #                  run in this particular order to ensure "pyrax" installs correctly.
   PYTHON_BIN="$(which python)"
-  virtualenv --python="${PYTHON_BIN}" /opt/test-maas
+  virtualenv --python="${PYTHON_BIN}" --no-pip /opt/test-maas
+  curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+  /opt/test-maas/bin/python get-pip.py
   /opt/test-maas/bin/pip install "jinja2" --isolated --upgrade --force-reinstall
```

Comment:
All these requirements being added into the venv are unconstrained. It may be best to build a working venv and use 'pip freeze' to construct a constraint file, then use the dependency update hook to propose weekly updates to the constraints file to keep them current.

With a constraints file you'd have less failures from external factors hitting the repo and would be able to manage their effects on important work being done.